### PR TITLE
feat: implement ERC-1167 minimal proxies

### PR DESCRIPTION
**Pull request by 0xble**

### CODE REVIEW

File: test/crowdfund/Crowdfund.t.sol [hunk 1]
```diff
@@ -1,12 +1,14 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8;
 
+import "forge-std/Test.sol";
```
Comment (arr00) [inline test/crowdfund/Crowdfund.t.sol:4]: not used

File: test/crowdfund/ReraiseETHCrowdfund.t.sol [hunk 2]
```diff
@@ -1,9 +1,11 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8;
 
+import "forge-std/Test.sol";
```
Comment (arr00) [inline test/crowdfund/ReraiseETHCrowdfund.t.sol:4]: no used

File: test/party/PartyGovernanceNFT.t.sol
Comment (arr00) [inline test/party/PartyGovernanceNFT.t.sol:1]: Why remove `PartyGovernanceNFTForkedTest`?

File: test/proposals/ListOnZoraProposalForked.t.sol [hunk 3]
```diff
@@ -229,58 +229,6 @@ contract ListOnZoraProposalForkedTest is SetupPartyHelper {
         assertEq(address(party).balance, proposalData.listPrice);
     }
 
-    function testForked_simpleZora() public onlyForked {
```
Comment (arr00) [inline test/proposals/ListOnZoraProposalForked.t.sol:232]: why remove this test?

File: test/proposals/ListOnZoraProposalIntegration.t.sol [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+import "../../contracts/globals/LibGlobals.sol";
+import "../../contracts/proposals/ProposalExecutionEngine.sol";
+import "../../contracts/distribution/TokenDistributor.sol";
+
+import "../TestUtils.sol";
+import "../DummyERC721.sol";
+import "./ZoraTestUtils.sol";
+import "../TestUsers.sol";
+
+contract ListOnZoraProposalIntegrationTest is Test, TestUtils, ZoraTestUtils {
```
Comment (arr00) [inline test/proposals/ListOnZoraProposalIntegration.t.sol:16]: We've moved to have all test that utilizes forks be in contracts suffixed with `ForkedTest`

Comment (0xble) [reply test/proposals/ListOnZoraProposalIntegration.t.sol:16]: There is already a `ListOnZoraProposalForked.t.sol` lol

**Comment by arr00** [comment]
There are still a few changes from #285 which are being reverted in this PR which seems to just be merge issues. These changes break the test suite. Any time that a forked test is used, it must be in a contract which has the suffix `ForkedTest`.

**Review by arr00 — CHANGES_REQUESTED** [review]
There are still a few changes from #285  which are being reverted in this PR which seems to just be merge issues. These changes break the test suite. Any time that a forked test is used, it must be in a contract which has the suffix `ForkedTest`.

File: test/proposals/ListOnZoraProposalIntegration.t.sol (hunk 4 above)
Comment (arr00) [reply test/proposals/ListOnZoraProposalIntegration.t.sol:16]: This test was integrated into `ListOnZoraProposalForked.t.sol` in #285

**Comment by 0xble** [comment]
> There are still a few changes from #285 which are being reverted in this PR which seems to just be merge issues. These changes break the test suite. Any time that a forked test is used, it must be in a contract which has the suffix `ForkedTest`.

I think it's fixed now 😩

File: test/crowdfund/ReraiseETHCrowdfund.t.sol [hunk 5]
```diff
@@ -244,36 +254,6 @@ contract ReraiseETHCrowdfundTest is LintJSON, TestUtils, ERC721Receiver {
         assertEq(crowdfund.pendingVotingPower(initialContributor), initialContribution);
     }
 
-    function test_initialContribution_aboveMaxTotalContribution() public {
-        address payable initialContributor = payable(_randomAddress());
```
Comment (arr00) [inline test/crowdfund/ReraiseETHCrowdfund.t.sol:248]: why delete this?

**Review by arr00 — APPROVED** [review]
Looks good

**Pull request merged**